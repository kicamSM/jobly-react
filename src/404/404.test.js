import React from 'react';
import { render, screen } from '@testing-library/react';
import NotFound from './404'
import { toMatchSnapshot } from 'jest-snapshot'

expect.extend({ toMatchSnapshot });

test('404 renders without crashing', () => {
  render(<NotFound />);
});

test('Not found message displays', () => {
    render(<NotFound />);

    const text  = screen.getByText("Sorry the page you were looking for doesn't exist.");
    expect(text).toHaveTextContent("Sorry the page you were looking for doesn't exist.");
});

test('Should display "404"', () => {
    render(<NotFound />);
  
    expect(screen.getByText('404')).toBeInTheDocument();
  });

test('matches snapshot', () => {
    const {asFragment} = render(<NotFound />);
    expect(asFragment()).toMatchSnapshot(); 
  })