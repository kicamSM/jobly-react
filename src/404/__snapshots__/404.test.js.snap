// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Not-Found card"
  >
    <div
      class="card-title"
    >
      <h1>
        404
      </h1>
    </div>
    <div
      class="card-body"
    >
      <p
        class="card-text"
      >
        Sorry the page you were looking for doesn't exist.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
